ui/app/components/app/permission-page-container/permission-page-container-content/permission-page-container-content.component.js

```diff
@@ -12,63 +12,69 @@ export default class PermissionPageContainerContent extends PureComponent {
     selectedPermissions: PropTypes.object.isRequired,
     permissionsDescriptions: PropTypes.object.isRequired,
     onPermissionToggle: PropTypes.func.isRequired,
-    selectedAccount: PropTypes.object,
+    selectedIdentities: PropTypes.array,
     redirect: PropTypes.bool,
     permissionRejected: PropTypes.bool,
   }
 
   static defaultProps = {
     redirect: null,
     permissionRejected: null,
-    selectedAccount: {},
+    selectedIdentities: [],
   }
 
   static contextTypes = {
     t: PropTypes.func,
   }
 
-  renderAccountInfo = (account) => {
+  renderRedirectVisual () {
+    const { permissionRejected, selectedIdentities, domainMetadata } = this.props
     return (
-      <div className="permission-approval-visual__account-info">
-        <div className="permission-approval-visual__account-info__label">
-          { account.label }
+      <div className="permission-approval-visual__redirect">
+        <IconWithFallBack icon={domainMetadata.icon} name={domainMetadata.name} />
+        <div className="permission-approval-visual__redirect-center">
+          { permissionRejected
+            ? <span className="permission-approval-visual__reject" ><i className="fa fa-times-circle" /></span>
+            : <span className="permission-approval-visual__check" />
+          }
+          <img className="permission-approval-visual__broken-line" src="/images/broken-line.svg" />
         </div>
-        <div className="permission-approval-visual__account-info__address">
-          { account.truncatedAddress }
+        <div className="permission-approval-visual__identicon-container">
+          <div className="permission-approval-visual__identicon-border">
+            <Identicon
+              className="permission-approval-visual__identicon"
+              address={selectedIdentities[0].address}
+              diameter={54}
+            />
+          </div>
         </div>
       </div>
     )
   }
 
+  renderNonRedirectVisual () {
+    const { requestMetadata, domainMetadata } = this.props
+
+    return (
+      <div>
+        <IconWithFallBack icon={domainMetadata.icon} name={domainMetadata.name} />
+        <h1>{domainMetadata.name}</h1>
+        <h2>{requestMetadata.origin}</h2>
+      </div>
```

Comment:
Nit: could you return a fragment here instead of a wrapper div? The div doesn't seem especially useful here.

```suggestion
      <>
        <IconWithFallBack icon={domainMetadata.icon} name={domainMetadata.name} />
        <h1>{domainMetadata.name}</h1>
        <h2>{requestMetadata.origin}</h2>
      </>
```

Reply:
This went away with aafa179f0